draft-ietf-jsonpath-base.md

```diff
@@ -1309,96 +1231,124 @@ Queries:
 {: title="Filter selector examples"}
 
-### List Selector
+## Appenders
 
-The list selector allows combining member names, array indices,
-slices, and filters in a single selector.
+A JSONPath query consists of root identifier followed by a sequence of appenders.
+Valid appenders are
 
-Note: The list selector was called "union selector" in
-{{JSONPath-orig}}, as it was intended to solve use cases addressed by
-the union selector in XPath.
-However, the term "union" has the connotation of a set operation that involves
-merging input sets while avoiding duplicates, so the concept was
-renamed into "list selector".
+  * Child appender `[<selectors>]`, where `<selectors>` is one or more of
+    several selector types, which match the nodes to select,
+    separated by commas
+  * Descendant child appender `..[<selectors>]`
+
+The descendant child appender is a child appender that has been made to
+recursively select values within nested objects and arrays by prepending it with `..`.
+
+
+### Child Appender
 
 #### Syntax
 {: unnumbered}
 
-The list selector is syntactically related to the
-`dot-selector`, `index-selector`, `slice-selector`, and the `filter-selector`.
-It contains two or more entries, separated by commas.
+The child appender has the form `[<selectors>]` where `<selectors>` is a comma-delimited
+collection of one or more selectors.
+Each selector is defined below.
 
 ~~~~ abnf
-list-selector  = "[" S list-entry 1*(S "," S list-entry) S "]"
+child-appender = "[" S selector 1*(S "," S selector) S "]"
 
-list-entry     =  ( quoted-member-name /
-                    element-index      /
-                    slice-index /
-                    filter
+selector       =  ( name-selector /
+                    index-selector      /
+                    slice-selector /
+                    filter-selector
                   )
 ~~~~
 
-#### Semantics
-{: unnumbered}
+Shorthand notations exist for child appenders with either:
 
-A list selector selects the nodes that are selected by
-the selector entries in the list and yields the concatenation of the
-lists (in the order of the selector entries) of nodes selected by the
-selector entries.
-Note that any node selected in more than one of the selector entries is kept
-as many times in the nodelist.
+* a single element that is a `wildcard`, as `wild-selector-shorthand`
+* a single element that is a `name-selector`, as `name-selector-shorthand`.
 
-To be valid, integer values in the `element-index` and `slice-index`
-components MUST be in the I-JSON {{-i-json}} range of exact values, see
-{{synsem-overview}}.
+This shorthand entirely replaces the bracketed syntax.
+For example, `['child']` and `.child` are equivalent as appenders,
+as are `[*]` and `.*`.
 
+A `name-selector-shorthand` starts with a dot `.` followed by an object member's name.
 
-#### Examples
+~~~~ abnf
+wild-selector-shorthand     = "." wildcard
+name-selector-shorthand    = "." dot-member-name
```

Comment:
The above productions are orphans. They need to be referenced in the rest of the ABNF.